File: internal/src/main/java/dev/mvmo/sicklang/internal/object/NullObject.java
package dev.mvmo.sicklang.internal.object;

public record NullObject() implements SickObject {

    public static NullObject NULL = new NullObject();

    @Override
    public String inspect() {
        return "null";
    }

    @Override
    public ObjectType objectType() {
        return ObjectType.NULL;
    }

    public static NullObject get() {
        return NULL;
    }

}

File: internal/src/main/java/dev/mvmo/sicklang/internal/object/NullObject.kt
package dev.mvmo.sicklang.internal.`object`

class NullObject : SickObject {
    override fun inspect() =
        "null"

    override fun objectType() =
        ObjectType.NULL

    companion object {
        @JvmField
        val NULL = NullObject()
        fun get() = NULL
    }
}